src/challenges.js

```diff
@@ -28,18 +36,22 @@ const cacheFunction = cb => {
 
 /* ======================== Recursion Practice ============================ */
 const reverseStr = str => {
-  // reverse str takes in a string and returns that string in reversed order
-  // The only difference between the way you've solved this before and now is that you need to do it recursivley!
+  var first = str[0];
+  var last = str[str.length -1];
+  var str1 = reverseStr(str.substring(1, str.length - 1));
+  return last + str1 + first;
 };
+// based on https://stackoverflow.com/questions/4859208/recursive-string-reversal-function-in-javascript
 
 const checkMatchingLeaves = obj => {
   // return true if every property on `obj` is the same
```

Comment:
I never was able to give a live lecture going over this function. I did, however, link a video out to our Lead TA explaining the solution. I'd recommend looking at it, as it may help you out next week!